packages/std/src/errors/system_error.rs

```diff
@@ -16,9 +16,10 @@ use crate::HumanAddr;
 #[serde(rename_all = "snake_case")]
 #[non_exhaustive]
 pub enum SystemError {
-    InvalidRequest { error: String },
+    InvalidRequest { msg: Vec<u8> },
+    InvalidResponse { msg: Vec<u8> },
```

Comment:
I imagine that the json contents will always be autogenerated when sent as part of the usual flows. If we're observing messages that can not be parsed as JSON it's probably because the message is completely mangled (memory bug, or wrong string being sent across) in which case it's more useful to just see the full string.

Reply:
I agree with the JSON aspect of it. But when you look at this aller code:

```rust
impl Querier for CustomQuerier {
    fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
        // parse into our custom query class
        let request: QueryRequest<CustomQuery> = match from_slice(bin_request) {
            Ok(v) => v,
            Err(e) => {
                return Err(SystemError::InvalidRequest {
                    error: format!("Parsing QueryRequest: {}", e),
                })
            }
        };
        if let QueryRequest::Custom(custom_query) = &request {
            Ok(execute(&custom_query))
        } else {
            self.base.handle_query(&request)
        }
    }
}
```

you see that it can as well be a musmatch in the generic type argument of


```rust
pub enum QueryRequest<T> {
    Bank(BankQuery),
    Custom(T),
    Staking(StakingQuery),
    Wasm(WasmQuery),
}
```

When there is a mismatch between the contract's expectation of the querier and the querier implementation, the serde error would provide hints on missing fields.

Reply:
That is true as well... Maybe it's best to keep the field as `error: String` and include both the error, and the original message?

Reply:
yes, that's what I suggested, but in 2 fields

Reply:
Oh yeah, that's a good point. Should the error be formatted, or kept as a `Box<dyn Error>` at this point? i think it'll be more efficient to only format it later, but we might lose the context of what we were trying to parse to begin with?

Reply:
Since SystemError can fully be serialized and deserialized, it should be an owned String. Thinking about this, `original` should be a `Binary` instead of `Vec<u8>` as this results in a compace base64 encoding instead of an array of integers (one per byte).

Reply:
Ah yes, good point. It shall be done :+1: